File: src/main/java/com/learn/EmployeeManagementSystem/service/PayrollService.java
package com.learn.EmployeeManagementSystem.service;

import com.learn.EmployeeManagementSystem.DTOs.requestDto.PayrollRequest;
import com.learn.EmployeeManagementSystem.DTOs.responseDTO.PayrollResponse;

import java.util.List;

public interface PayrollService {

    List<PayrollResponse> getAllPayrolls();

    PayrollResponse getPayrollById(Long id);

    PayrollResponse createPayroll(PayrollRequest payrollRequest);

    PayrollResponse updatePayroll(Long id, PayrollRequest payrollRequest);

    void deletePayroll(Long id);
}

File: frontend/frontend/src/services/payrollService.js
import api from './api';

export const payrollService = {
  // GET /payrolls
  getAllPayrolls: async () => {
    const response = await api.get('/payrolls');
    return response.data;
  },

  // GET /payrolls/{id}
  getPayrollById: async (id) => {
    const response = await api.get(`/payrolls/${id}`);
    return response.data;
  },

  // POST /payrolls
  createPayroll: async (payrollData) => {
    const response = await api.post('/payrolls', payrollData);
    return response.data;
  },

  // PUT /payrolls/{id}
  updatePayroll: async (id, payrollData) => {
    const response = await api.put(`/payrolls/${id}`, payrollData);
    return response.data;
  },

  // DELETE /payrolls/{id}
  deletePayroll: async (id) => {
    const response = await api.delete(`/payrolls/${id}`);
    return response.data;
  },
};

export default payrollService;
